src/transport_ofi.c

```diff
@@ -1137,6 +1185,35 @@ int shmem_transport_init(void)
     shmem_transport_ofi_contexts = malloc(shmem_transport_ofi_avail_ctx
                                           * sizeof(shmem_transport_ctx_t*));
 
+    /* STX max settings */
+    shmem_transport_ofi_stx_max = shmem_internal_params.OFI_STX_MAX;
+
+    /* STX sharing settings */
+    char *type = shmem_internal_params.OFI_STX_SHARE_ALGORITHM;
+    if (0 == strcmp(type, "round-robin")) {
+        shmem_transport_stx_share_alg = ROUNDROBIN;
+    } else if (0 == strcmp(type, "random")) {
+        shmem_transport_stx_share_alg = RANDOM;
+    } else {
+        RAISE_WARN_MSG("Ignoring bad STX share algorithm '%s'\n", type);
+    }
+
+    if (shmem_internal_thread_level == SHMEM_THREAD_SINGLE &&
+        shmem_transport_ofi_stx_max > 1) {
+        /* We need only 1 STX per PE with SHMEM_THREAD_SINGLE */
+        RAISE_WARN_MSG("Ignoring invalid STX max setting '%ld' w/THREAD_SINGLE, using 1\n", 
+                       shmem_transport_ofi_stx_max);
+        shmem_transport_ofi_stx_max = 1;
```

Comment:
If the default is more than 1, I think we will get this warning with every THREAD_SINGLE run.  You can look at ```shmem_internal_params.OFI_STX_SHARE_ALGORITHM_provided``` to see if the user provided a value versus if the variable was set to the default value.

I think we also want to land in this case for FUNNELED.  Maybe SERIALIZED, although, I'm not sure..